utils/misc.ml

```diff
@@ -17,6 +17,13 @@ exception Fatal_error
 let fatal_error msg =
   prerr_string ">> Fatal error: "; prerr_endline msg; raise Fatal_error
 
+let fatal_errorf fmt =
+  let (_ : string) = Format.flush_str_formatter () in
+  Format.kfprintf (fun ppf ->
+      fatal_error (Format.flush_str_formatter ()))
+    Format.str_formatter
```

Comment:
> We need Format.kasprintf...

Fortunately, It's not terribly hard to implement, see [here](https://github.com/dbuenzli/fmt/blob/9b7910318e71c42b87416838b2aa1b2c2ad064db/src/fmt.ml#L16-L19).